### PR TITLE
Move forward to Maven 3 only

### LINKED ISSUE #139: Final Move to Maven 3+ Only
* Change Maven Version 3.0 minimum
* Updated dependencies
* Using maven-artifact-transfer 0.9.0
* Replace old code with generics code.

### PR DESCRIPTION
**Pull request by epabst**
Fixes #139

### CODE REVIEW

**Comment by epabst** [comment]
Would you like my commits squashed with your commits?

**Comment by epabst** [comment]
I took the liberty of doing so.  I also reverted some minor changes such as import ordering to make merge conflicts easier for other changes.

File: pom.xml [hunk 1]
```diff
@@ -284,6 +290,11 @@
       <version>1.9.5</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.maven</groupId>
+      <artifactId>maven-compat</artifactId>
+      <version>3.0</version>
+    </dependency>
```
Comment (khmarbaise) [inline pom.xml]: This should not been used cause it contains the compatibility layer to Maven 2...so the plugin should work without this one...

Comment (epabst) [reply pom.xml]: I found that quite difficult.  Can you help?

Comment (dannil) [reply pom.xml]: @epabst I assume @khmarbaise means that maven-compat dependency can be dropped; it's a library which supplies Maven 2-specific classes, which I guess this plugin uses, making it able to run under Maven 3 as well. If the plugin is supposed to drop support for Maven 2 that dependency can be safely removed.

**Comment by epabst** [comment]
Without that dependency,  it won't compile.   I don't know how to make it
compile without it although I tried.

Eric

File: pom.xml (hunk 1 above)
Comment (epabst) [reply pom.xml]: Without that dependency,  it won't compile.   I don't know how to make it
compile without it although I tried.

Comment (dannil) [reply pom.xml]: I've tried removing the dependency myself and it seems the biggest offender (among others) is WagonManager. According to https://maven.apache.org/ref/3.5.0/apidocs/org/apache/maven/artifact/manager/WagonManager.html, the whole class has been marked for deprecation; you'll therefore need to find a replacement class which fills the same purpose. A starting point is https://maven.apache.org/ref/3.5.0/apidocs/org/apache/maven/repository/legacy/WagonManager.html, it seems to be the 3.0 equivalent.

Comment (dannil) [reply pom.xml]: Another alternative could be to refactor the code to use the [Maven Wagon](http://maven.apache.org/wagon/index.html) library with the HTTP provider, but it entails a lot of refactoring. I suggest you wait for input from @khmarbaise.

Comment (khmarbaise) [reply pom.xml]: If I correctly see the problem you need maven-artifact-transfer which handled the resolving of artifacts See https://maven.apache.org/shared/maven-artifact-transfer/  but I think one functionality is missing in maven-artifact-transfer to get the list of existing versions of a dependency..(Which I'm currently working on..)...

Comment (famod) [reply pom.xml]: @khmarbaise

> but I think one functionality is missing in maven-artifact-transfer to get the list of existing versions of a dependency..(Which I'm currently working on..)...

Is this done?

**Pull request closed without merging**